src/plugins/ab-testing.js

```diff
@@ -0,0 +1,59 @@
+import experiments from '../experiments';
+
+export default function(ctx, inject) {
+  // is there an experiment?
+  if (experiments.length === 0) {
+    inject('experiments', () => {
+      return false;
+    });
+    return false;
+  }
+  // check consent
+  const consentCookie = ctx.$cookies.get('klaro');
+  if (!consentCookie?.matomo) {
```

Comment:
The matomo AB testing cookie might/should be separate from the matomo analytics cookie. So this check should change once we do the set up with the plugin